### PR TITLE
Fix assembly block issue

### PR DESCRIPTION
**Pull request by pythonberg1997**

### CODE REVIEW

File: contracts/SlashIndicator.sol [hunk 1]
```diff
@@ -190,19 +190,15 @@ contract SlashIndicator is ISlashIndicator,System,IParamSubscriber, IApplication
...
-    require(srcNumA+256 > block.number && srcNumB+256 > block.number, "too old block involved in the evidence");
+    require(_evidence.voteA.srcNum+256 > block.number && _evidence.voteB.srcNum+256 > block.number, "too old block involved in the evidence");
     require(!(_evidence.voteA.srcHash == _evidence.voteB.srcHash &&
     _evidence.voteA.tarHash == _evidence.voteB.tarHash), "two identical votes");
-    require(srcNumA < tarNumA && srcNumB < tarNumB, "source number bigger than target number");
+    require(_evidence.voteA.srcNum < _evidence.voteA.tarNum && _evidence.voteB.srcNum < _evidence.voteB.tarNum, "source number bigger than target number");
 
     // Vote rules check
-    if (!((srcNumA<srcNumB && tarNumB<tarNumA) || (srcNumB<srcNumA && tarNumA<tarNumB)) && tarNumA != tarNumB) {
+    if (!((_evidence.voteA.srcNum<_evidence.voteB.srcNum && _evidence.voteB.tarNum<_evidence.voteA.tarNum) || (_evidence.voteB.srcNum<_evidence.voteA.srcNum && _evidence.voteA.tarNum<_evidence.voteB.tarNum)) && _evidence.voteA.tarNum != _evidence.voteB.tarNum) {
```
Comment (NashBC) [inline contracts/SlashIndicator.sol]: if (!((_evidence.voteA.srcNum<=_evidence.voteB.srcNum && _evidence.voteB.tarNum<_evidence.voteA.tarNum) || (_evidence.voteB.srcNum<=_evidence.voteA.srcNum && _evidence.voteA.tarNum<_evidence.voteB.tarNum)))

The source number can be equal. The tarNumA != tarNumB seems redundant?

File: contracts/SlashIndicator.sol [hunk 2]
```diff
@@ -220,57 +216,45 @@ contract SlashIndicator is ISlashIndicator,System,IParamSubscriber, IApplication
...
-      TypesToBytes.bytes32ToBytes(32, _evidence.voteB.tarHash, cur);
-      input = abi.encodePacked(input, cur);
-      input = abi.encodePacked(input, _evidence.voteB.sig);
-      input = abi.encodePacked(input, voteAddr);
-    }
+    bytes memory output = new bytes(1);
+
+    bytes memory pre = new bytes(32);
+    bytes memory cur = new bytes(32);
+    TypesToBytes.uintToBytes(32, srcNum, pre);
```
Comment (NashBC) [inline contracts/SlashIndicator.sol]: pre and cur here is not very clear, can we give a better name?

File: contracts/SlashIndicator.sol (hunk 1 above)
Comment (pythonberg1997) [reply contracts/SlashIndicator.sol]: There are too many logical judgment so it's not easy to understand. I rewrite this part.

File: contracts/SlashIndicator.sol (hunk 2 above)
Comment (pythonberg1997) [reply contracts/SlashIndicator.sol]: Done

File: contracts/SlashIndicator.sol [hunk 3]
```diff
@@ -190,21 +190,19 @@ contract SlashIndicator is ISlashIndicator,System,IParamSubscriber, IApplication
...
-    require(srcNumA < tarNumA && srcNumB < tarNumB, "source number bigger than target number");
+    require(_evidence.voteA.srcNum < _evidence.voteA.tarNum &&
+      _evidence.voteB.srcNum < _evidence.voteB.tarNum, "source number bigger than target number");
 
     // Vote rules check
-    if (!((srcNumA<srcNumB && tarNumB<tarNumA) || (srcNumB<srcNumA && tarNumA<tarNumB)) && tarNumA != tarNumB) {
-      revert(string(abi.encodePacked("no violation of vote rules")));
-    }
+    require((_evidence.voteA.srcNum<_evidence.voteB.srcNum && _evidence.voteB.tarNum<_evidence.voteA.tarNum) ||
+    (_evidence.voteB.srcNum<_evidence.voteA.srcNum && _evidence.voteA.tarNum<_evidence.voteB.tarNum) ||
```
Comment (NashBC) [inline contracts/SlashIndicator.sol]: The second line needs to be indented， and the same to the third line.

Comment (pythonberg1997) [reply contracts/SlashIndicator.sol]: Fixed

**Review by realuncle — APPROVED** [review]

**Review by NashBC — APPROVED** [review]

**Pull request merged**